README.md

```diff
@@ -87,6 +92,103 @@ If an OCM provider exposes the capability `/mfa-capable`, it indicates that it w
+  * 'algorithm' specify the algorithm used to generate signature
+  * 'headers' specify the properties used when generating the signature
+  * 'signature' the signature of an array containing the properties listed in 'headers'. Some properties like content-length, date, digest, and host are mandatory to protect against authenticity override.
+
+
+### How to generate the Signature for outgoing request
+
+After properties are set in the headers, the Signature is generated and added to the list.  
+
+This is a quick PHP example of headers for outgoing request:
+
+```php
+    $headers = [
+        '(request-target)' => 'post /path',
+        'content-length' => strlen($payload),
+        'date' => gmdate('D, d M Y H:i:s T'),
+        'digest': 'SHA-256=' . base64_encode(hash('sha256', utf8_encode($payload), true)),
+        'host': 'hostname.of.the.recipient',
+    ];
+
+    openssl_sign(implode("\n", $headers), $signed, $privateKey, OPENSSL_ALGO_SHA256);
+		
+    $signature = [
+        'keyId' => 'https://author.hostname/key',
+        'algorithm' => 'ras-sha256',
+        'headers' => 'content-length date digest host',
+        'signature' => $signed 
+    ];
+
+    $headers['Signature'] = implode(',', $signature);
+```
+
+
+### How to confirm Signature on incoming request
+
+The first step would be to confirm the validity of each properties:
+
+- '(request-target)' and 'host' are immutable to the type of the request and the local/current host,
+- 'content-length' and 'digest' can be re-generated and compared from the payload of the request,
+- A maximum TTL must be applied to 'date' and current timestamp, 
+- regarding data extracted from 'Signature':
+  * using 'keyId' to get the public key from remote signatory,
+  * 'headers' is used to generate the clear version of the signature and must contain at least 'content-length', 'date', 'digest' and 'host',
+  * 'signature' is the encrypted version of the signature.
+
+Using the clear version of the signature, the public key and the encrypted version, it is easy to confirm the authenticity of the request:
+
+```php
+    $clear = [
+        '(request-target)' => 'post /path',
+        'content-length' => strlen($payload),
+        'date' => 'Mon, 08 Jul 2024 14:16:20 GMT',
+        'digest': 'SHA-256=' . base64_encode(hash('sha256', utf8_encode($payload), true)),
+        'host': $localhost
+    ];
+    
+    $signed = "DzN12OCS1rsA[...]o0VmxjQooRo6HHabg==";
+    if (openssl_verify(implode("\n", $clear), $signed, $publicKey, 'sha256') !== 1) {
+        throw new InvalidSignatureException('signature issue');
+    }
+```
+
+### Validating the payload
+
+while signed, it is still needed to confirm the validity of the payload.  
+The last step is to ensure that the payload implies action initiated on the behalf the source of the request.  
+
+As an example, if the payload is about initiating a new share the file owner has to be an account from the instance at the origin of the request.  
```

Comment:
Apart the minor rephrasing I proposed above, I'm not sure this can actually be done in general, and the example in particular does not seem feasible: how can an EFSS validate something that refers to user belonging to a remote EFSS? Ultimately it has to trust the remote EFSS and its payload.

Reply:
Sure, but I think what is meant is: if the owner is specified as `alice@cernbox.cern.ch` but the sender is `cloud.pondersource.com` then the receiving server should flag the domain name mismatch.

We could also make this automatic in a future version of the spec, so that owner would just be `alice` and the `@cernbox.cern.ch` would be implied silently from the keyId of the signature.